### PR TITLE
Feature: Add support to load custom themes in Theme builder

### PR DESCRIPTION
**Pull request by shubhamofbce**
## Description

I have updated the existing load theme method such that it can also accept the theme object to load. I will fetch the theme object from hub and load it using the same method.

Closes: #7687

The change is deployed on my hf space. https://huggingface.co/spaces/shubhamofbce/theme-color-fix?logs=container
This space also has the changes included in #7756

## Tests

- [x] Ran tests
- [x] Formatted backend using ruff.

### CODE REVIEW

**Comment by shubhamofbce** [comment]
@abidlabs @aliabd Another PR ready for review. You can try the change on my space link mentioned in the description.

**Comment by abidlabs** [comment]
Taking a look now, thanks @shubhamofbce

File: gradio/themes/builder_app.py [hunk 1]
```diff
@@ -99,6 +99,20 @@ def get_doc_theme_var_groups():
...
                     )
                     load_theme_btn = gr.Button("Load Theme", elem_id="load_theme")
+                    gr.Markdown(
+                        """
+                        Load a custom theme. Just copy any theme name from [Theme Gallery](https://huggingface.co/spaces/gradio/theme-gallery)
+                        """
+                    )
+
+                    custom_theme_name = gr.Textbox(
+                        label="THeme Name",
```
Comment (abidlabs) [inline gradio/themes/builder_app.py:109]:
```suggestion
                        label="Theme Name",
```

**Comment by abidlabs** [comment]
@shubhamofbce I'm not able to get custom loading to work for all themes. For example, I tried `gstaff/xkcd`, the most popular custom theme, with your Space and it did not load the theme. Can you take a look?

**Comment by shubhamofbce** [comment]
@abidlabs I just tried that theme and it worked fine. I think you must have tried it in dark mode where the impact of that theme is less visible. Please try it one more time and let me know if it still doesn't works.
https://github.com/gradio-app/gradio/assets/45156638/5eab100d-08f4-4bb0-b8ac-fa0c2a077cea
[video](https://github.com/gradio-app/gradio/assets/45156638/5eab100d-08f4-4bb0-b8ac-fa0c2a077cea)

**Comment by shubhamofbce** [comment]
@abidlabs Any updates on this.

**Comment by shubhamofbce** [comment]
@abidlabs We can merge this, it is not a breaking change and working smooth on hf space. It will be a helpful feature for our users.

**Comment by abidlabs** [comment]
Going to just cc @aliabd to give a final review

**Comment by abidlabs** [comment]
@shubhamofbce just getting a chance to revisit this. I'm still seeing the issue that I described before, where custom themes are not fully loaded. Even in the video that you included, when load `gstaff/xkcd`, the colors load but the font of the theme does not. Compare:

image

to:

image

The styling is very different. Can we fix this issue and then I'm happy to review again

**Comment by abidlabs** [comment]
Going to close this for now @shubhamofbce but feel free to reopen once the issue above has been addressed.

**Pull request closed without merging**